clang/test/Sema/warn-extra-cast-function-type-strict.c

```diff
@@ -0,0 +1,42 @@
+// RUN: %clang_cc1 %s -fsyntax-only -Wextra -verify
```

Comment:
As written, tests checks only default language mode for C (which is `gnu99` AFAIK). Maybe we want to check other modes as well?

Reply:
add runs for -std=cNN for all remaining 5 of them? and also add extra test cases where conflicts arise?

Reply:
Yes, more `RUN` lines with different `-std=` options, and potentially more `expected` directives tailored for particular language mode. We have an example of this in C++ defect report tests: https://github.com/llvm/llvm-project/blob/main/clang/test/CXX/drs/dr6xx.cpp

Note that I might be biased here, as someone who've spent a significant time with C++ defect report test suite.